microservices/user-facing-api/controllers/userAnswers/uploadUserQuestions.js

```diff
@@ -0,0 +1,61 @@
+import * as userAnswerDao from "./useranswer-dao.js";
+import * as userDao from "../users/user-dao.js";
+import { logger, LogLevel } from "../../utils/logging.js";
+import { MinioClient } from "../../utils/minio.js";
+import fs from 'fs/promises';
+import dotenv from 'dotenv';
+dotenv.config();
+
+const minioClient = new MinioClient();
+
+const uploadUserQuestions = async (req, res) => {
+    try {
+        const userId = req.params.userId;
+        if (!userId) {
+            return res.status(400).send({ message: "User ID is required" });
+        }
+        const user = await userDao.findUserById(userId);
+        if (!user) {
+            return res.status(404).send({ message: "User not found" });
+        }
+        let questionAnsObj = [];
+        let newQuestionAnsObj = [];
+
+        //called from microservice to upload new questions
+        if(req.body.questionAnsObj.length>0){
+            newQuestionAnsObj = req.body.questionAnsObj;
+        } else {
+        const localFilePath = "../../static/questions.json";
```

Comment:
Updating logic based on who's calling this function is not a good idea. Why does this function need to know about fetching questions from MinIO? This logic should be outside of this function and every call to this function should have the questionsAnsObj.